src/openapi/streaming/streaming.js

```diff
@@ -83,11 +83,39 @@ function setNewContextId() {
     }
 }
 
+/**
+ * Find matching delay based on current retry count/index.
+ * @param retryLevels - The retry levels that contain different delays for various retry count levels.
+ *                      Structure: [ { level: Number, delay: Number } ].
+ * @param retryIndex - The current retry index/try/count.
+ * @param defaultDelay {number} - The default delay.
+ * @returns {number} Matching delay to retry index/try/count.
+ */
+function findRetryDelay(retryLevels, retryIndex, defaultDelay) {
+    let lastFoundDelay = defaultDelay;

```

Comment:
what if it gets past the levels - it shouldn't stay on the last item?

Reply:
It does. Current implementation uses last matching level and it's doesn't use exact matching. Instead it uses >=.

I've added full unit test coverage for all cases for both this method and streaming disconnect->retry lifecycle. 

This specif case is covered here:
https://github.com/SaxoBank/openapi-clientlib-js/pull/113/files#diff-30a8f287d3c66269a64c11675a8ce443R146